tests/functional/aws-node-sdk/test/bucket/updateBucketQuota.js

```diff
@@ -0,0 +1,56 @@
+const AWS = require('aws-sdk');
+const S3 = AWS.S3;
+
+const assert = require('assert');
+const getConfig = require('../support/config');
+const sendRequest = require('../quota/tooling').sendRequest;
+
+const bucket = 'updatequotatestbucket';
+const nonExistantBucket = 'updatequotatestnonexistantbucket';
+const quota = { quota: 2000 };
+const negativeQuota = { quota: -1000 };
+const wrongquotaFromat = '1000';
+
+describe('Test update bucket quota', () => {
+    let s3;
+
+    before(() => {
+        const config = getConfig('default', { signatureVersion: 'v4' });
+        s3 = new S3(config);
+        AWS.config.update(config);
+    });
+
+    beforeEach(done => s3.createBucket({ Bucket: bucket }, done));
+
+    afterEach(done => s3.deleteBucket({ Bucket: bucket }, done));
+
+    it('should update the quota', done => {
+        sendRequest('PUT', '127.0.0.1:8000', `/${bucket}/?quota=true`, JSON.stringify(quota), err => {
+            assert.strictEqual(err, null);
+            done();
+        });
+    });
+
+    it('should return no such bucket error', done => {
+        sendRequest('PUT', '127.0.0.1:8000', `/${nonExistantBucket}/?quota=true`, JSON.stringify(quota), err => {
+            assert.strictEqual(err.Error.Code[0], 'NoSuchBucket');
+            done();
+        });
+    });
+
+    it('should return error when quota is negative', done => {
+        sendRequest('PUT', '127.0.0.1:8000', `/${bucket}/?quota=true`, JSON.stringify(negativeQuota), err => {
+            assert.strictEqual(err.Error.Code[0], 'InvalidArgument');
+            assert.strictEqual(err.Error.Message[0], 'Quota Value should be a positive number');
+            done();
+        });
+    });
+
+    it('should return error when quota is not in correct format', done => {
```

Comment:
Can we also try with a very big number? Just thinking about how we store it in mongodb, int32 will have low limits, just to be sure it works when the limit will be in TB (or PB)

Reply:
done here : https://github.com/scality/cloudserver/pull/5552/commits/eeeae66ade78935c691dc25b6a0717de25ffe2f3#diff-92ecfdca6f0579cab12b6c586b092bd6a9ada97335e47b83d0921462c2bd60ddR13